packages/antd/src/components/buttons/refresh/index.tsx

```diff
@@ -25,36 +28,45 @@ export const RefreshButton: React.FC<RefreshButtonProps> = ({
     resourceNameOrRouteName: propResourceNameOrRouteName,
     recordItemId,
     hideText = false,
-    meta,
-    metaData,
     dataProviderName,
     children,
     onClick,
+    meta: _meta,
+    metaData: _metaData,
     ...rest
 }) => {
     const translate = useTranslate();
 
-    const { identifier, id } = useResource(
+    const queryClient = useQueryClient();
+    const invalidates = useInvalidate();
+
+    const { resources, identifier, id } = useResource(
         resourceNameFromProps ?? propResourceNameOrRouteName,
     );
 
-    const { refetch, isFetching } = useOne({
-        resource: identifier,
-        id: recordItemId ?? id,
-        queryOptions: {
-            enabled: false,
-        },
-        meta: pickNotDeprecated(meta, metaData),
-        metaData: pickNotDeprecated(meta, metaData),
-        liveMode: "off",
-        dataProviderName,
+    const isInvalidating = !!queryClient.isFetching({
+        queryKey: queryKeys(
+            identifier,
+            pickDataProvider(identifier, dataProviderName, resources),
+        ).detail(recordItemId ?? id),
     });
 
+    const handleInvalidate = () => {
+        invalidates({
+            id: recordItemId ?? id,
+            invalidates: ["detail"],
+            dataProviderName,
+            resource: identifier,
+        });
+    };
+
     return (
         <Button
-            // TODO: fix any type
-            onClick={(e) => (onClick ? onClick(e as any) : refetch())}
-            icon={<RedoOutlined spin={isFetching} />}
+            onClick={(e) => {
+                onClick?.(e);
+                handleInvalidate();
+            }}
```

Comment:
```suggestion
            onClick={(e) => (onClick ? onClick(e) : refetch())}
```

This may be caused an issue, for users who previously passed `onClick`.